examples/plot_picard_ode.py

```diff
@@ -0,0 +1,103 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Anderson acceleration in application to Picard–Lindelöf theorem.
+================================================================
+
+With `Picard–Lindelöf theorem
+<https://en.wikipedia.org/wiki/Picard%E2%80%93Lindel%C3%B6f_theorem>`_ we can reduce differential equation solving to fixed point computations and simple integration.
+More precisely consider the ODE:
+
+.. math::
+
+  y'(t)=f(t,y(t))
+
+of some time-dependant dynamic :math:`f:\mathbb{R}\times\mathbb{R}^d\rightarrow\mathbb{R}^d` and initial conditions :math:`y(0)=y_0`. Then we have:
+
+.. math::
+
+  y(t)=y_0+\int_0^t f(s,y(s))\mathrm{d}s
+
+Then we can define the sequence of functions :math:`(\phi_k)` with :math:`\phi_0=0` recursively as follow:
+
+.. math::
+
+  \phi_{k+1}(t)=y_0+\int_0^t f(s,\phi_k(s))\mathrm{d}s
+
+Such sequence converges :math:`\lim_{k\rightarrow\infty}\phi_k=y` to the solution of the ODE.  
+  
+In this example we take :math:`f(t,y(t))=1+y(t)^2` with solution :math:`y(t)=\tan{t}`. 
+We used `scipy.integrate.cumtrapz` to perform integration, but any other numerical scheme could have worked.
+"""
+
+
+import jax
+import jax.numpy as jnp
+
+from jaxopt import AndersonAcceleration
+
+from jaxopt import objective
+from jaxopt.tree_util import tree_scalar_mul, tree_sub
+
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn import datasets
+from matplotlib.pyplot import cm
+import scipy.integrate
+
+
+#Solve y'(t)=1+t^2 differential equation, with solution y(t)=tan(t)
+def f(ti, phi):
+  return 1 + phi ** 2
+
+"""Fixed point iteration in the Picard method.
+See: https://en.wikipedia.org/wiki/Picard%E2%80%93Lindel%C3%B6f_theorem"""
+def integrate(phi_cur, ti, y0, dx):
+  f_phi = f(ti, phi_cur)
+  phi_next = scipy.integrate.cumtrapz(f_phi, initial=y0, dx=dx)
+  return phi_next
+
+y0 = 0
+num_interpolating_points = 400
+t0 = jnp.array(0.)
+tmax = 0.95 * (jnp.pi / 2) # stop before pi/2 to ensure convergence
+dx = (tmax - t0) / (num_interpolating_points-1)
+phi0 = jnp.zeros(num_interpolating_points)
+ti = np.linspace(t0, tmax, num_interpolating_points)
+
+sols = [phi0]
+aa = AndersonAcceleration(integrate, history_size=3, maxiter=50, ridge=1e-5, jit=False)
+sol, state = aa.init(phi0, ti, y0, dx)
+sols.append(sol)
+for k in range(aa.maxiter):
+  sol, state = aa.update(phi0, state, ti, y0, dx)
+  sols.append(sol)
+res = sols[-1] - np.tan(ti)
+print(f'Error of {jnp.linalg.norm(res)} with ground truth tan(t)')
+
+sols = sols[:8]
```

Comment:
Personal taste: I find this code a bit too packed. I would add some blank lines to let the code "breathe". 

I see mainly three blocks in this code: initialization of the figure, building the plot and setting up visual properties.